middle_end/flambda/to_cmm/un_cps_helper.ml

```diff
@@ -143,7 +143,11 @@ let and_ = binary Cmm.Cand
 let or_ = binary Cmm.Cor
 let xor_ = binary Cmm.Cxor
 
-let eq = binary Cmm.(Ccmpi Ceq)
+let eq ?(dbg=Debuginfo.none) x y =
+  match x, y with
+  | Cmm.Cconst_int (n, _), Cmm.Cop(Csubi, [Cconst_int (m, _); c], _) ->
+    binary Cmm.(Ccmpi Ceq) ~dbg c (Cmm.Cconst_int (m - n, dbg))
+  | _, _ -> binary Cmm.(Ccmpi Ceq) ~dbg x y
 let neq = binary Cmm.(Ccmpi Cne)
```

Comment:
Please add a blank line above this to differentiate the code for `eq` and `neq`

Reply:
Maybe add also add a comment stating why it's not really pertinent to apply the same optimization to `neq` (or add it to the comment for `eq`): since the optimization target is a pattern generated by `un_cps` for switches that are if/then/else , while other let's say "regular" (or at least flambda2-generated) uses of `eq` or `neq` are presumably already optimized by flambda2 if possible (or will be at one point)